### PR TITLE
switch dropdown menu to react-aria-components

### PR DESCRIPTION
**Pull request by jivey**
https://openstax.atlassian.net/browse/DISCO-29

The react-aria-components version of the menu is needed by the new responsive sidebar nav in assignments. The existing one gets cut off by the overflow style, this one renders outside the container.

### CODE REVIEW

File: src/components/__snapshots__/Tooltip.spec.tsx.snap [hunk 1]
```diff
@@ -85,7 +85,7 @@ Array [
       role="tooltip"
       style={
         Object {
-          "maxHeight": undefined,
+          "maxHeight": "100vh",
```
Comment (jivey) [inline src/components/__snapshots__/Tooltip.spec.tsx.snap:88]: I think this is coming from upgrading react-aria-components

**Comment by Dantemss** [comment]
Wait what happened to the button? Seems like it's all gone

**Comment by jivey** [comment]
I'm switching it to use the menu component from react-aria-components. The original version won't work with the new scroll overflow:
image

The button is still there, but because we get all the accessibility stuff like keyboard nav included with the RAC component, most of the code can be deleted. It should visually and functionally appear the same:
https://openstax.github.io/ui-components/refs/heads/dropdown-menu-updates/142d623f6b6256ab6ba00fd563a88dfc04fd9252/?story=dropdown-menu--primary

**Review by Dantemss — APPROVED** [review]
I feel like you could keep the keyboard navigation test to make sure it actually does what it says it does, but sure...

**Comment by jivey** [comment]
Yeah I wanted to keep the tests at first but they majorly broke and I don't know how much I want to test the internals of an already tested library on the consuming side.

**Comment by Dantemss** [comment]
Next time we release ui-components, let's ask Brooke to regression test the accessibility of the menu.

**Pull request merged**